### ISSUE TITLE
Should vue/order-in-components be enforced for mixins?

### ISSUE THREAD

**Issue by gburning** [vuejs/eslint-plugin-vue#813]
**Tell us about your environment**
* **ESLint version:** 5.13.0
* **eslint-plugin-vue version:** 5.1.0
* **Node version:** 11.6.0

**Please show your full configuration:**

```js
module.exports = {
  'extends': [
    'eslint:recommended',
    'standard',
    'plugin:vue/recommended'
  ],
  'rules': {
    'indent': ['error', 2],
    'vue/script-indent': ['error', 2, {
      'baseIndent': 1
    }],
    "vue/html-self-closing": ["error", {
      "html": {
        "void": "never",
        "normal": "never",
        "component": "always"
      },
      "svg": "always",
      "math": "always"
    }]
  },
  'overrides': [
    {
      'files': ['*.vue'],
      'rules': {
        'indent': 'off'
      }
    }
  ]
}
```

**What did you do? / What did you expect to happen? / What actually happened?**
I ran eslint on a mixin expecting the [`vue/order-in-components`](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/order-in-components.md) rule to be enforced. It does not seem to be however (I can put `props` after `computed` for example). So my question is if this is intended or if it is because of a bug/fault in my configuration?

**Comment by mysticatea** [comment]
Thank you for this issue.

Would you put both the expected behavior and the actual behavior with source code?

**Comment by gburning** [comment]
Sure, here's a simplified example:

**mixin.js**
```js
export default {
  methods: {
    update (value) {
      this.$emit('input', value)
    }
  },
  props: {
    label: {
      type: String,
      default: null
    },
    placeholder: {
      type: String,
      default: ''
    },
    value: {
      type: [String, Number],
      default: null
    }
  },
  computed: {
    pHolder () {
      return this.placeholder || this.label
    }
  }
}
```

**form-input.vue**
```vue
<template>
  <div>
    <label
      v-if="label"
    >
      {{label}}
    </label>
    <input
      :type="'text'"
      :placeholder="pHolder"
      :value="value"
      @input="update($event.target.value)"
    >
  </div>
</template>

<script>
  import mixin from 'mixin.js'

  export default {
    mixins: [mixin]
  }
</script>
```

Now, when I run the linter on the .vue file I'd expect an error to be thrown, since I'm putting my `methods` before my `props` in the mixin that extends the component. The end result is technically that the order of the properties in the component is wrong.

Let me know if more info is needed.

**Comment by mysticatea** [comment]
Okay, thank you for the explanation.

This is the expected behavior because ESLint checks each file individually. This means the check of a file doesn't affect another file.

You can put `//@vue/component` comment into the `mixin.js` to tell ESLint that the object has component properties.

> https://mysticatea.github.io/vue-eslint-demo/#eJyNUsFOwzAM/RUrQuqQGiZx7IkjJy7jxjh0jcuCUidykmqw9d9Jm64UISEuifwcPz8/5ywaq1BUYruFhz7itrGds4QU9oQnZzmAwraOJsB5TwAdhqNVvsoRQHSqDgibvjYRb68oQDhqf3eDnQ6bQpOLoSghv8kvhvEayvF0bN03oakPaKoV0YfDCnaBNb2VV3CWVAFFY2bCOelM3eDRGoX8b5ai+MkxCf1V/TKXw1PsDsivf6r5Hm80NAZUC597nNTBZmUXY4hM2bXVBHC5ZGxyZU29p0GUgqNBL6qzmDfXpb1JpRmboPu01ftyyrz7k4wevexr9gtKVjaW2ujTULKXreV06nbJW04KpCa5fImxdiiFJpWCnf7MHXL4nDxK38i7JD4pc6kTcgLQO0YUwxdpPscd

Further reading:

- https://eslint.vuejs.org/user-guide/#how-eslint-detects-components

**Comment by gburning** [comment]
Ok, that makes sence. Thanks!
